base/libgit2/reference.jl

```diff
@@ -169,10 +169,8 @@ function lookup_branch(repo::GitRepo,
     err = ccall((:git_branch_lookup, :libgit2), Cint,
```

Comment:
I asked on the libgit2 slack, and they said that if an allocating function returns an error then you don't need to free the resulting pointer. So you can just replace this by an `@check ccall(...)`